tests/rptest/tests/redpanda_kerberos_test.py

```diff
@@ -193,3 +194,82 @@ def test_license_nag(self):
         wait_until(self._has_license_nag,
                    timeout_sec=self.LICENSE_CHECK_INTERVAL_SEC * 2,
                    err_msg="License nag failed to appear")
+
+
+class RedpandaKerberosRulesTesting(RedpandaKerberosTestBase):
+    def __init__(self, test_context, **kwargs):
+        super(RedpandaKerberosRulesTesting, self).__init__(test_context,
+                                                           num_nodes=3,
+                                                           **kwargs)
+
+    @cluster(num_nodes=3)
+    @parametrize(rules=["RULE:[1:$1test$0](client.*)", "DEFAULT"],
+                 kerberos_principal="client",
+                 rp_users=["client", f"clienttest{REALM}"],
+                 expected_topics=["restricted", "always_visible"],
+                 acl=[("restricted", f"clienttest{REALM}"),
+                      ("dontseeme", "client"), ("always_visible", "*")])
+    @parametrize(rules=[
+        "RULE:[2:$1testbad$0](client.*)",
+        "RULE:[1:$1testgood](client.*)s/client(.*)/$1redpanda/U", "DEFAULT"
+    ],
+                 kerberos_principal="client",
+                 rp_users=["TESTGOODREDPANDA", "no"],
+                 expected_topics=["restricted", "always_visible"],
+                 acl=[("restricted", "TESTGOODREDPANDA"), ("no", "no"),
+                      ("always_visible", "*")])
```

Comment:
I couldn't get too crazy.  If I did ducktape would be unable to create the directory (as the name would be too big).

Reply:
Due to the parameters getting into the filename?

It's all unit tested anyway, right? So this is just ensuring that the wiring is working?

Reply:
> Due to the parameters getting into the filename?

Yes

> It's all unit tested anyway, right? So this is just ensuring that the wiring is working?

Yes